src/osx/carbon/metafile.cpp

```diff
@@ -92,12 +92,13 @@ wxMetafileRefData::wxMetafileRefData(CFDataRef data) :
 wxMetafileRefData::wxMetafileRefData( const wxString& filename )
 {
     Init();
-
+#ifndef __WXOSX_IPHONE__
```

Comment:
It looks like it should be possible to avoid this, as the code is there since 2008 and must have compiled under iOS until f83577df451. Again, knowing what the exact problem was would be helpful here.

Reply:
problem here is 
../src/osx/carbon/metafile.cpp:99:31: error: use of undeclared identifier 'wxOSXCreateURLFromFileSystemPath'
        wxCFRef<CFURLRef> url(wxOSXCreateURLFromFileSystemPath(filename));

wxOSXCreateURLFromFileSystemPath is only declared in include/wx/osx/core/private.h:WXDLLIMPEXP_BASE CFURLRef wxOSXCreateURLFromFileSystemPath( const wxString& path);
but in src/common/filefn.cpp the rule is the following. 
#if defined(__WXMAC__) && !defined(__WXOSX_IPHONE__)

any hint to solve this ??

Reply:
Sorry for not being clear: my idea is that `wxOSXCreateURLFromFileSystemPath()` should be made available under iOS too, as it's just a wrapper for `CFURLCreateWithFileSystemPath()` which seems to be available there. So I'd rename it to something (@csomor what prefix do you use for functions available under both macOS and iOS?) and move it outside of `!wxOSX_USE_IPHONE` check.

Reply:
unfortunately there is no common name prefix in use. I've used `wxOSX` due to the common ancestry for all platforms.

Reply:
I'll make this available in core, thus for all platforms,

Reply:
in eaec76ee97435a36a295d7f79139e747cbd6d45e